primitives/src/pow.rs

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Proof-of-work related integer types.
+
+use core::fmt;
+
+use crate::parse::{self, PrefixedHexError, UnprefixedHexError};
+
+/// Encoding of 256-bit target as 32-bit float.
+///
+/// This is used to encode a target into the block header. Satoshi made this part of consensus code
+/// in the original version of Bitcoin, likely copying an idea from OpenSSL.
+///
+/// OpenSSL's bignum (BN) type has an encoding, which is even called "compact" as in bitcoin, which
+/// is exactly this format.
+///
+/// # Note on order/equality
+///
+/// Usage of the ordering and equality traits for this type may be surprising. Converting between
+/// `CompactTarget` and `Target` is lossy *in both directions* (there are multiple `CompactTarget`
+/// values that map to the same `Target` value). Ordering and equality for this type are defined in
+/// terms of the underlying `u32`.
+#[derive(Copy, Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct CompactTarget(u32);
+
+impl CompactTarget {
+    /// Creates a `CompactTarget` from a prefixed hex string.
+    pub fn from_hex(s: &str) -> Result<Self, PrefixedHexError> {
+        let target = parse::hex_u32_prefixed(s)?;
+        Ok(Self::from_consensus(target))
+    }
+
+    /// Creates a `CompactTarget` from an unprefixed hex string.
+    pub fn from_unprefixed_hex(s: &str) -> Result<Self, UnprefixedHexError> {
+        let target = parse::hex_u32_unprefixed(s)?;
+        Ok(Self::from_consensus(target))
+    }
```

Comment:
These are non-trivial conversions that can be implemented using other methods so I think they belong to the extension trait.

Reply:
Just to make sure I fully understand your position; these functions introduce logic that is complex enough to potentially be problematic to stabalization and they can be handrolled by folks that only use `primitives` anyways so better to put them in the extension trait?

Reply:
Yes, especially the error types are the main reason here.

Reply:
Cool, got it. Will re-spin. Thanks